dap_aria_mapping/pipeline/novelty/openalex_topic_novelty.py

```diff
@@ -0,0 +1,94 @@
+""" 
+Script to calculate novelty scores at topic level, using OpenAlex papers
+Uses typer to create a command line interface
+
+Usage examples:
+
+All levels, full dataset:
+python dap_aria_mapping/pipeline/novelty/openalex_topic_novelty.py
+
+One level, test dataset:
+python dap_aria_mapping/pipeline/novelty/openalex_topic_novelty.py --taxonomy-level 1
+"""
+from dap_aria_mapping import logging, PROJECT_DIR, BUCKET_NAME
+from dap_aria_mapping.getters.novelty import (
+    get_openalex_novelty_scores,
+    get_openalex_topic_pair_commonness,
+)
+import dap_aria_mapping.utils.novelty_utils as nu
+from dap_aria_mapping.getters.taxonomies import get_topic_names
+from nesta_ds_utils.loading_saving.S3 import upload_obj
+import typer
+
+OUTPUT_DIR = "outputs/novelty"
+
+
+def calculate_topic_novelty(
+    taxonomy_level: int = 0,
+    from_local: bool = False,
+    save_to_local: bool = False,
+    min_pair_counts: int = 50,
+    min_doc_counts: int = 50,
+):
+    """
+    Calculate novelty scores for topics using OpenAlex data and uploads them on s3
+
+    Args:
+        taxonomy_level (int, optional): Taxonomy level to use for novelty calculation. Must be between 1 and 5. If set to 0, uses all levels
+        from_local (bool, optional): Whether to use data from local disk
+        save_to_local (bool, optional): Whether to also save the novelty scores to local disk
+        min_pair_counts (int, optional): Minimum number of times a topic pair must appear in the corpus to be considered for novelty calculation (for the score based on topic pairs)
+        min_doc_counts (int, optional): Minimum number of times a topic must appear in the corpus to be considered for including into the analysis outputs
+    """
+    if taxonomy_level == 0:
+        levels = list(range(1, 6))
+    else:
+        levels = [taxonomy_level]
+    # Loop over taxonomy levels
+    for level in levels:
+        logging.info(f"Processing taxonomy level {level}")
+        # Fetch novelty scores for works
+        work_novelty_df = get_openalex_novelty_scores(
+            level=level, from_local=from_local
+        ).drop_duplicates("work_id")
+        # Fetch commonness values for topic pairs
+        topic_pairs_df = get_openalex_topic_pair_commonness(
+            level=level, from_local=from_local
+        )
+        # Fetch topic names
+        topic_names = get_topic_names(
+            taxonomy_class="cooccur", name_type="entity", level=level
```

Comment:
FYI, the code that will be merged from #60 allows you to get the chatgpt-generated names using this same function, setting `name_type` argument to "chatgpt" and specifying the set of entities it uses to query chatgpt via the argument `n_top` (set it to 35). It returns a list of four-item tuples, which can be quickly transformed to a pandas df.

Reply:
Very nice, can add an issue then to change the `name_type` argument once #60 is merged!